includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -409,6 +417,14 @@ public function get_upe_available_payment_methods() {
 		$available_payment_methods = [];
 
 		foreach ( self::UPE_AVAILABLE_METHODS as $payment_method_class ) {
+			//if merchant is outside US, Link payment method should not be available
+			$account = WC_Stripe::get_instance()->account;
+			if (
+				WC_Stripe_UPE_Payment_Method_Link::class === $payment_method_class &&
+				( ! isset( $account->get_cached_account_data()['country'] ) || 'US' !== $account->get_cached_account_data()['country'] )
+			) {
+				continue;
+			}
```

Comment:
Maybe it's better to have this part of the `UPE_Payment_Method` class, returning `true` by default, but having a special condition in `WC_Stripe_UPE_Payment_Method_Link`?

Reply:
I like the approach @dmvrtx suggested. We can also make use of the `supported_countries` property in `WC_Stripe_UPE_Payment_Method_Link` and use it in this condition.

Reply:
changed in [ca7a146](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2452/commits/ca7a146738af657437e38aa109d602f8ed5c5270). thank you!